### PR TITLE
Marvl3 layers jndi

### PR DESCRIPTION
**Pull request by ggalibert**
forked https://github.com/aodn/geoserver-config/pull/257 

Modified store to use a JNDI entry.

To do when marvl3 geonetwork record on RC/Prod:
1/ Add links to metadata records in WMS layers.

### CODE REVIEW

**Comment by lbesnard** [comment]
Looking at it now. I have to investigate a bit more, but so far
 * the schema exists on db rc, but is empty
 * data is only on dbprod. So harvest_read has to be updated in po box  (/var/lib/tomcat7/default/conf/Catalina/localhost/geoserver.xml )  to point to dbprod.
 * there is only a *_data layer, no *_map

**Comment by lbesnard** [comment]
@danfruehauf I'm confused, doing a 
```bash
grep -rnw '.../chef-private/' -e "marvl3"
```
I actually find entries which creates the schema and role on both RC/123. But there is no talend json file containing any information related to marvl3. Although data exists on RC but no prod. So something managed to write there but ... HELP !

**Comment by lbesnard** [comment]
> data is only on dbprod. So harvest_read has to be updated in po box

sorry, this is wrong

**Comment by lbesnard** [comment]
can't display anything on my pobox. Tried to do a cql filtering, but same

**Comment by danfruehauf** [comment]
@lbesnard  As explained, this schema is solely on dbprod - been restored once after the work done by your team. Can be restored on dbrc if we wish to do so.

Why the geoserver layers don't work - I will have no idea.

**Comment by lbesnard** [comment]
well I created a sql view in geoserver with a limit of the first 500 rows. It works now. Is the full table going to work on RC ??

**Comment by ggalibert** [comment]
We're not going to display everything anyway. There will always be a filtering on time and depth at least.

**Comment by lbesnard** [comment]
By default, everything is displayed right now. It would be up to the user to select whatever filter from the dropdown menu.

**Comment by ggalibert** [comment]
@lbesnard could you please review again with the latest changes?

**Comment by lbesnard** [comment]
@ggalibert in terms of speed, it is the same right ?

**Comment by ggalibert** [comment]
Still slow on po box indeed. But try subsetting like :

http://10.11.12.13/geoserver/imos/wms?service=WMS&version=1.1.0&request=GetMap&layers=imos:marvl3_atlas_tempn_map&styles=&bbox=145,-44.3,154,-35&width=512&height=436&srs=EPSG:4326&format=application/openlayers

**Comment by lbesnard** [comment]
yeap that's much better. Got the 500 error too

**Comment by ggalibert** [comment]
Getting HTTP Status 500 - even though content ftl looks good. Similar as other collection. Even more simplified, any time we try to access some property from a feature we get the error. At least on PO box. Keen to merge to master though and look at what's happening on RC.

**Comment by lbesnard** [comment]
I'm trying editing the ftl

**Comment by ggalibert** [comment]
Do not merge to production yet please

**Comment by lbesnard** [comment]
why mergin ? it's not working

File: workspaces/imos/JNDI_marvl3/marvl3_atlas_psaln_map/content.ftl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<div class="feature">
+<#list features as feature>
+
+<#if (feature_index < 50) >
+<div class="featurewhite">
+<b>Bin longitude and latitude :</b> ${feature.LONGITUDE_bin.value} - ${feature.LATITUDE_bin.value}<br/>
```
Comment (lbesnard) [inline workspaces/imos/JNDI_marvl3/marvl3_atlas_psaln_map/content.ftl]: if you look at this http://10.11.12.13/geoserver/web/?wicket:bookmarkablePage=:org.geoserver.web.data.resource.ResourceConfigurationPage&name=marvl3_atlas_tempn_map&wsName=imos

you'll see this doesnt exist in the table. thats why it's not working

**Comment by lbesnard** [comment]
@ggalibert there are errors with the ftl, it works well on mine if I remove a few lines and keep this 
```
<div class="feature">
<#list features as feature>

<#if (feature_index < 50) >
<div class="featurewhite">
<b>Time start and end:</b> ${feature.TIME_bound_min.value?date("dd/MM/yyyy")} - ${feature.TIME_bound_max.value?date("dd/MM/yyyy")}<br/>
<b>Depth min and max :</b> ${feature.DEPTH_bound_min.value} - ${feature.DEPTH_bound_max.value}<br/>
<BR>
</div>
</#if>
</#list>
</div>
```

**Comment by lbesnard** [comment]
this works better
```
<div class="feature">
<#list features as feature>

<#if (feature_index < 50) >
<div class="featurewhite">
<b>Bin longitude and latitude :</b> ${feature.LONGITUDE_bound_max.value} - ${feature.LATITUDE_bound_min.value}<br/>
<b>Time start and end:</b> ${feature.TIME_bound_min.value?date("dd/MM/yyyy")} - ${feature.TIME_bound_max.value?date("dd/MM/yyyy")}<br/>
<b>Depth min and max :</b> ${feature.DEPTH_bound_min.value} - ${feature.DEPTH_bound_max.value}<br/>
<b>Number of temperature measurements :</b> ${feature.TEMP_n.value}<br/>
<b>Temperature mean ± SD (°C):</b> <#if feature.TEMP_n.value == '1.0'> ${feature.TEMP_mean.value} ± NA <#else> ${feature.TEMP_mean.value} ± ${feature.TEMP_stddev.value}</#if><br/>
<b>Temperature range  (°C):</b> ${feature.TEMP_min.value} - ${feature.TEMP_max.value}<br/>
<BR>
</div>
</#if>
</#list>
</div>
~         
```

**Comment by lbesnard** [comment]
same thing should be done for the salt layer

**Comment by ggalibert** [comment]
Your example doesn't work for me...

**Comment by ggalibert** [comment]
For some reason have to do :
```
<div class="feature">
<#list features as feature>

<#if (feature_index < 50) >
<div class="featurewhite">
<b>Bin longitude and latitude :</b> ${feature.LONGITUDE_bin.value} - ${feature.LATITUDE_bin.value}<br/>
<b>Time start and end:</b> ${feature.TIME_bound_min.rawValue?string("dd/MM/yyyy")} - ${feature.TIME_bound_max.rawValue?string("dd/MM/yyyy")}<br/>
<b>Depth min and max :</b> ${feature.DEPTH_bound_min.value} - ${feature.DEPTH_bound_max.value}<br/>
<b>Number of temperature measurements :</b> ${feature.TEMP_n.value}<br/>
<b>Temperature mean ± SD (°C):</b> <#if feature.TEMP_n.value == '1.0'> ${feature.TEMP_mean.rawValue?string("0.00")} ± NA <#else> ${feature.TEMP_mean.rawValue?string("0.00")} ± ${feature.TEMP_stddev.rawValue?string("0.00")}</#if><br/>
<b>Temperature range  (°C):</b> ${feature.TEMP_min.rawValue?string("0.00")} - ${feature.TEMP_max.rawValue?string("0.00")}<br/>
<BR>
</div>
</#if>
</#list>
</div>
```

**Comment by lbesnard** [comment]
Are you sure it's workign ?
LONGITUDE_bin.value should be LONGITUDE_bound_min

**Pull request merged**